File: main/binary-tree-preorder-traversal/binary-tree-preorder-traversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<int> PreorderTraversal(TreeNode root) {
        var path = new List<int>();
        if (root == null) return path;

        var stack = new Stack<TreeNode>();
        for (stack.Push(root); stack.Count != 0; ) {
            root = stack.Pop();
            path.Add(root.val);
            if (root.right != null) stack.Push(root.right);
            if (root.left != null) stack.Push(root.left);
        }

        return path;
    }
}

File: main/binary-tree-preorder-traversal/binary-tree-preorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def preorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        path = []
        if root is None:
            return path

        stack = []
        stack.append(root)

        while stack:
            root = stack.pop()
            path.append(root.val)
            if root.right is not None:
                stack.append(root.right)
            if root.left is not None:
                stack.append(root.left)

        return path
